src/panels/config/lovelace/resources/dialog-lovelace-resource-detail.ts

```diff
@@ -199,21 +199,23 @@ export class DialogLovelaceResourceDetail extends LitElement {
 
   private async _updateResource() {
     this._submitting = true;
-    try {
-      const values: LovelaceResourcesMutableParams = {
-        url: this._url.trim(),
-        res_type: this._type,
-      };
-      if (this._params!.resource) {
-        await this._params!.updateResource(values);
-      } else {
-        await this._params!.createResource(values);
+    if (this._type) {
```

Comment:
```
    if (!this._type) {
    	return;
    }
```

Reply:
Why? I'd assume we do not want to try to create or update a resource if no type has been set.

Reply:
Exactly. So we want to return as soon as possible. The if statement makes it unnecessary less readable.

Reply:
Also, move it above `this._submitting = true;`

Reply:
> 
> 
> Exactly. So we want to return as soon as possible. The if statement makes it unnecessary less readable.

Oh man, my bad 🙄 . Did not see that you wrote `return`. I was too quick and only focused on the condition itself which of course then made no sense in my mind... 🤦